src/rust/aggregator/src/data/user_history.rs

```diff
@@ -376,7 +403,7 @@ async fn aggregate_fill<'a>(
                     END
                 END,
                 last_updated_at = $4
-            WHERE order_id = $2 AND market_id = $3;
+            WHERE order_id = $2 AND market_id = $3
```

Comment:
With the recent deduping logic updates, I think this is off

Basically this function only gets called for the fill event emitted to the maker, but the changes need to be applied to the taker as well

So yes we should dedupe fill events by only calling this function for when the maker address is the address that the event was emitted to

But the taker *and* maker orders, defined by their order IDs as included in the fill event, need to be updated

Reply:
It also looks like this handles swaps but not market orders?

Reply:
I understand that anytime a fill event is applied to a swap the status is updated to closed, but this should only happen when remaining size is 0

Reply:
Fix in #525